src/Widgets/DataBreadcrumbWidget/DataBreadcrumbWidgetComponent.tsx

```diff
@@ -0,0 +1,73 @@
+import {
+  DataItem,
+  LinkTag,
+  provideComponent,
+  useData,
+  useDataLocator,
+} from 'scrivito'
+import { DataBreadcrumbWidget } from './DataBreadcrumbWidgetClass'
+import { ensureString } from '../../utils/ensureString'
+
+provideComponent(DataBreadcrumbWidget, ({ widget }) => {
+  const currentItem = useData().dataItem()
+  const parentAttributeName = useDataLocator(widget.get('parentData'))
+    .dataItemAttribute()
+    ?.attributeName()
+  const labelAttributeName = useDataLocator(widget.get('data'))
+    .dataItemAttribute()
+    ?.attributeName()
+
+  if (!currentItem) {
+    return <nav aria-label="breadcrumb" />
+  }
+
+  const breadcrumbItems: DataItem[] = []
+
+  for (
+    let ancestorItem: unknown = currentItem, loopGuardIds = new Set<string>();
+    ancestorItem instanceof DataItem && !loopGuardIds.has(ancestorItem.id());
+    ancestorItem = parentAttributeName
+      ? ancestorItem.get(parentAttributeName)
+      : null
+  ) {
+    breadcrumbItems.unshift(ancestorItem)
+    loopGuardIds.add(ancestorItem.id())
+  }
```

Comment:
I find this for loop hard to understand. I would prefer a more verbose variant, that is (hopefully) easier to understand, e.g.

```ts
function breadcrumbItems(
  currentItem: DataItem | null,
  parentAttributeName?: string,
): DataItem[] {
  if (!currentItem) return []
  if (!parentAttributeName) return [currentItem]

  const breadcrumbItems: DataItem[] = []
  const loopGuardIds = new Set()

  let ancestorItem: unknown = currentItem
  while (
    ancestorItem instanceof DataItem &&
    !loopGuardIds.has(ancestorItem.id())
  ) {
    breadcrumbItems.unshift(ancestorItem)
    loopGuardIds.add(ancestorItem.id())
    ancestorItem = ancestorItem.get(parentAttributeName)
  }

  return breadcrumbItems
}
```